src/phase0/state_transition/epoch_processing.rs

```diff
@@ -5,9 +5,9 @@ use crate::phase0::state_transition::{
     get_block_root_at_slot, get_current_epoch, get_previous_epoch, get_randao_mix,
     get_total_active_balance, get_total_balance, get_validator_churn_limit, increase_balance,
     initiate_validator_exit, is_active_validator, is_eligible_for_activation,
-    is_eligible_for_activation_queue, Context, Error,
+    is_eligible_for_activation_queue, Checkpoint, Context, Error,
```

Comment:
I noticed `Copy` isn't derived for `Checkpoint` nor, mostly, any of the other types in this project. Just wondering what the design rationale is there (for my own curiosity/knowledge purposes).

Reply:
`Copy` should only be derived for types that are "lightweight" to copy

we could add `Copy` to `Checkpoint` but it is more uniform to just derive `Clone` -- we could go and make a separate pass to restrict types that derive Clone to deriving Copy but i don't know if it will gain a lot at the moment

we definitely want to use `Clone` when the type allocates, like if it includes a List or Vector SSZ type